lightning/src/ln/channel.rs

```diff
@@ -293,6 +293,18 @@ struct HTLCStats {
 	pending_htlcs_value_msat: u64,
 	on_counterparty_tx_dust_exposure_msat: u64,
 	on_holder_tx_dust_exposure_msat: u64,
+	holding_cell_msat: u64,
+}
+
+/// An enum gathering stats on commitment transaction, either local or remote.
+struct CommitmentStats<'a> {
+	tx: CommitmentTransaction, // the transaction info
+	feerate_per_kw: u32, // the feerate included to build the transaction
+	total_fee_sat: u64, // the total fee included in the transaction
+	num_nondust_htlcs: usize,  // the number of HTLC outputs (dust HTLCs *non*-included)
+	htlcs_included: Vec<(HTLCOutputInCommitment, Option<&'a HTLCSource>)>, // the list of HTLCs whic were not ignored when building the trnasacition
+	local_balance_msat: u64,
```

Comment:
Could use a doc comment here and on the next saying that this is the balance *before fees* but considering dust limits. Obviously we should clean this up and pass the fee as a parameter to build_commitment_transaction, but I'm okay with that being a followup.

Reply:
Added a comment and bookmarked the suggestion in follow-up issue.